DALOptimizer/MatchExpr.cs

```diff
@@ -0,0 +1,265 @@
+                        script.RemoveText(startOffset, Len);
+                        script.InsertText(startOffset, "DummyText");
+                    }
+                }
+                if (Pat.varDeclMthd().Match(expr).Success)
+                {
+                    if (expr.Parent.GetType().Name == "BlockStatement" &&
+                        expr.Parent.Parent.GetType().Name == "MethodDeclaration")
+                    {
+                        var varName = expr.FirstChild.NextSibling.FirstChild.GetText();
+                        script.Replace(expr, Pat.varDeclMthd1(varName));
+                    }
+                }
+            }
+
+        }
+        private void ExprStatement(DocumentScript script, CSharpFile file)
+        {
+            foreach (var expr in file.IndexOfExprStmt)
+            {
+                var copy = (ExpressionStatement)expr.Clone();
+
+                AllPatterns Pat = new AllPatterns();
+                if (Pat.FillExpr().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.StoredProc().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.sqlConnstmt().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.ConnOpenExprStmt().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.ConnCloseExprStmt().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.CmdDisposeExprStmt().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.ConvertToInt32().Match(expr).Success) { script.Remove(expr, true); continue; }
+                if (Pat.ExNonQuery().Match(expr).Success)
+                {
+                    string varName = expr.FirstChild.FirstChild.GetText();
+                    string objName = expr.FirstChild.LastChild.FirstChild.FirstChild.GetText();
+                    var expr1 = Pat.ExeStrdProc(varName, objName);
+                    script.Replace(expr, expr1);
+                    continue;
+                }
+
+                if (Pat.SqlDataAdapterExprStmt().Match(expr).Success)
+                {
+                    string retType = expr.GetParent<MethodDeclaration>().ReturnType.GetText();
+                    if (retType == "DataTable" || retType == "System.Data.DataTable")   //
+                    {
+                        string varName = expr.GetText().Split("()".ToCharArray())[1];
+                        var getDtTbl = Pat.GetDtTbl(varName);
+                        script.Replace(expr, getDtTbl);
+                    }
+                    else if (retType == "DataSet" || retType == "System.Data.DataSet")
+                    {
+                        string varName = expr.GetText().Split("()".ToCharArray())[1];
+                        var getDtSet = Pat.GetDtSet(varName);
+                        script.Replace(expr, getDtSet);
```

Comment:
Saw similar code in `VarDeclarationStmt` method as well. Can we refactor to avoid duplication?